router.php

```diff
@@ -49,16 +50,8 @@
     if (false === $target) {
         throw new RuntimeException('FUNCTION_TARGET is not set');
     }
-    if (!is_callable($target)) {
-        throw new InvalidArgumentException(sprintf(
-            'Function target is not callable: "%s"',
-            $target
-        ));
-    }
-
-    $signatureType = getenv('FUNCTION_SIGNATURE_TYPE', true) ?: 'http';
 
-    $invoker = new Invoker($target, $signatureType);
```

Comment:
Do we not need to pass in the `$signatureType` to preserve backwards compatibility here? ie GCFv1 customers today might be using non-declarative cloudevent functions in PHP

Reply:
It still respects the env var, I just moved it from being read here to being read within `router.php` so I could test it properly. `router.php` is hard to test because it's being used as script instead of a just declaring a function or class.

Reply:
There are a few reasons why this is not a good idea

1. This breaks backwards compatibility
2. The decision to inject the environment variable was intentional. This is a best practice (see dependency injection), as it keeps the logic of where the value comes from (e.g. the environment) separate from the class which uses the value (e.g. the Invoker)
3. Having the value be injectable is _more testable_, as you can test the class without having to mess with environment variables.